concordium-consensus/src/Concordium/GlobalState/Persistent/LMDB.hs

```diff
@@ -172,25 +245,28 @@ dbStepSize = 2^(26 :: Int) -- 64MB
 dbInitSize :: Int
 dbInitSize = dbStepSize
 
-
 -- NB: The @ati@ is stored in an external database if chosen to.
 
 -- |Initialize database handlers in ReadWrite mode.
 -- This simply loads the references and does not initialize the databases.
-databaseHandlers :: RuntimeParameters -> IO (DatabaseHandlers pv st)
-databaseHandlers RuntimeParameters{..} = makeDatabaseHandlers rpTreeStateDir False
+-- The initial size is set to 64MB.
+databaseHandlers :: FilePath -> IO (DatabaseHandlers pv st)
+databaseHandlers treeStateDir = makeDatabaseHandlers treeStateDir False dbInitSize
 
 -- |Initialize database handlers.
+-- The size will be rounded up to a multiple of 'dbStepSize'.
 makeDatabaseHandlers
   :: FilePath
   -- ^Path of database
   -> Bool
   -- ^Open read only
+  -> Int
+  -- ^Initial database size
   -> IO (DatabaseHandlers pv st)
-makeDatabaseHandlers treeStateDir readOnly = do
+makeDatabaseHandlers treeStateDir readOnly initSize = do
   _storeEnv <- mdb_env_create
-  mdb_env_set_mapsize _storeEnv dbInitSize
-  mdb_env_set_maxdbs _storeEnv 4
+  mdb_env_set_mapsize _storeEnv (initSize + dbStepSize - initSize `mod` dbStepSize)
```

Comment:
Is the idea of this that the initial size is a multiple of step size?

It would be useful to be clear about why that is important, so that this is not changed in future updates.